### PR TITLE
Add JUnit XML output to `ec validate image`

### PR DESCRIPTION
**Pull request by zregvart**
What constitutes a valid or useful JUnit XML output is debatable. There are several XML Schemas describing the format that differ in smaller details. There is also xUnit and Open Test Reporting XML formats.

This adds support for a variant of JUnit XML report that allows for multiple `testsuite`s, with each `testsuite` corresponding to one image or component.

To reach similar fidelity as other formats certain values are stored in `testsuite` `property` elements, both messages and `failure` `error` content is utilized.

Acceptance test is added that loosely verifies the output to be as expected.

Ref. https://issues.redhat.com/browse/HACBS-1971

### CODE REVIEW

File: internal/applicationsnapshot/junit.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	"sort"
+	"strings"
+
+	conftestOutput "github.com/open-policy-agent/conftest/output"
+	"golang.org/x/exp/maps"
+	"k8s.io/kubernetes/test/utils/junit"
+)
+
+type testSuites struct {
+	XMLName    xml.Name           `xml:"testsuites"`
```
[the remark this answers is not in the document]
Comment (zregvart) [reply internal/applicationsnapshot/junit.go:31]: The XML structure looks like this:

```xml
<testsuites>
  <testsuite />
  <testsuite />
  <!-- ... -->
</testsuites>
```

We only have the `junit.TestSuite` (the `<testsuite>`) element from k8s.io/kubernetes/test/utils/junit

File: internal/applicationsnapshot/junit.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	}
+
+	name := fmt.Sprintf("%s%s", r.Message, desc)
+
+	if code, ok := r.Metadata["code"].(string); ok {
+		name = fmt.Sprintf("%s: %s", code, name)
+	}
+
+	return &junit.TestCase{
+		Name:      name,
```
[the remark this answers is not in the document]
Comment (zregvart) [reply internal/applicationsnapshot/junit.go:74]: Most of this is guesswork, not sure what reporting tools require and the mapping is not exactly 1:1

File: features/validate_image.feature [hunk 3]
```diff
@@ -855,3 +855,30 @@ Feature: evaluate enterprise contract
...
...(localhost:\d+\/acceptance\/image@sha256:[0-9a-f]{64}\)" tests="2" errors="0" failures="1" time="0" timestamp="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{8,9}Z" hostname="">(<property name="image" value="localhost:\d+\/acceptance\/image@sha256:[0-9a-f]{64}"><\/property>|<property name="key" value="-----BEGIN PUBLIC KEY-----[^"]+"><\/property>|<property name="success" value="false"><\/property>|<property name="keyId" value=""><\/property>|<property name="signature" value="[a-zA-Z0-9+\/]+={0,2}"><\/property>|<property name="metadata.predicateType" value="https:\/\/slsa.dev\/provenance\/v0.2"><\/property>|<property name="metadata.type" value="https:\/\/in-toto.io\/Statement\/v0.1"><\/property>|<property name="metadata.predicateBuildType" value="https:\/\/tekton.dev\/attestations\/chains\/pipelinerun@v2"><\/property>)+<testcase name="main.acceptor: Pass" classname="main.acceptor: Pass" time="0"><\/testcase><testcase name="main.rejector: Fails always \[effective_on=2022-01-01T00:00:00Z\]" classname="main.rejector: Fails always \[effective_on=2022-01-01T00:00:00Z\]" time="0"><failure message="Fails always" type=""><!\[CDATA\[Fails always\]\]><\/failure><\/testcase><\/testsuite><\/testsuites>
```
Comment (simonbaird) [inline features/validate_image.feature:883]: I guess we can't easily indent for readability here, right?

**Review by simonbaird — APPROVED** [review]

File: features/validate_image.feature (hunk 3 above)
Comment (zregvart) [reply features/validate_image.feature:883]: The output currently is not indented, so in the .feature file where we don't parse the XML (as we do with JSON) it has to be un-indented as well.

**Pull request merged**